[mm_logs] make aten mm info readable

YUNQIUGUO opened the pull request:
Summary:
as title. make it into a table like


e.g. also see pic in test plan

| Name     | M   | N   | K   | Count |
| aten.mm | 16  | 6   |  16 |     1     |
...

Test Plan: {F1975907876}
<img width="1090" alt="Screenshot 2025-03-11 at 3 13 00 PM" src="https://github.com/user-attachments/assets/ffae8c56-e32c-49cc-bbfb-5b8d216b8657" />


Differential Revision: D70825664




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang commented:
don't use tabulate. Its not in https://github.com/pytorch/pytorch/blob/main/requirements.txt

Skylion007 commented on torch/_inductor/compile_fx.py:
```diff
@@ -834,10 +834,16 @@ def _compile_fx_inner(
+    from tabulate import tabulate
+
+    mm_table_data = []
+    for key, value in counters["aten_mm_info"].items():
+        name, m, n, k = key.split("_")
+        mm_table_data.append([name, m, n, k, value])
     log.info(
-        "Overview info of inductor aten mms: %s",
-        ", ".join(
-            f"({key}: {value})" for key, value in counters["aten_mm_info"].items()
+        "Overview info of inductor aten mms:\n%s",
+        tabulate(
+            mm_table_data, headers=["Name", "M", "N", "K", "Count"], tablefmt="grid"
         ),
```
Constructing this table is pretty expensive potentially. Let's only do it if needed:
```suggestion
    if log.isEnabledFor(logging.INFO):
        from tabulate import tabulate

        mm_table_data = []
        for key, value in counters["aten_mm_info"].items():
            name, m, n, k = key.split("_")
            mm_table_data.append([name, m, n, k, value])
        log.info(
            "Overview info of inductor aten mms:\n%s",
            tabulate(
                mm_table_data, headers=["Name", "M", "N", "K", "Count"], tablefmt="grid"
            ),
        )

YUNQIUGUO replied:
sure! that makes sense to me! will fix

henrylhtsang commented:
can you place a sample logs?

otherwise LGTM

henrylhtsang reviewed (approved):
preemptive approved

YUNQIUGUO committed:
[mm_logs] make aten mm info readable (#148800)

Summary:

as title. make it into a table like


e.g.
| Name     | M   | N   | K   | Count |
| aten.mm | 16  | 6   |  16 |     1     |
...


also see pic in test plan

Test Plan: {F1975975567}

Reviewed By: henrylhtsang

Differential Revision: D70825664

ZainRizvi commented:
@pytorchbot merge -f "landed internally"
